dali/pipeline/operator/eager_operator.h

```diff
@@ -187,15 +210,15 @@ EagerOperator<Backend>::Run(
     const std::vector<std::shared_ptr<TensorList<InBackend>>> &inputs,
     const std::unordered_map<std::string, std::shared_ptr<TensorList<CPUBackend>>> &kwargs,
     int batch_size) {
-  return Run(inputs, kwargs, shared_cuda_stream, batch_size);
+  return Run(inputs, kwargs, GetSharedCudaStream(), batch_size);
 }
 
 template <>
 std::vector<std::shared_ptr<TensorList<CPUBackend>>> EagerOperator<CPUBackend>::Run(
     const std::vector<std::shared_ptr<TensorList<CPUBackend>>> &inputs,
     const std::unordered_map<std::string, std::shared_ptr<TensorList<CPUBackend>>> &kwargs,
     int batch_size) {
-  return Run(inputs, kwargs, shared_thread_pool.get(), batch_size);
+  return Run(inputs, kwargs, GetSharedThreadPool().get(), batch_size);
```

Comment:
GetSharedThreadPool() is shared_ptr, but once the (inner) Run is called we dispose that shared_ptr, is thar right? So what will happen if someone updates the pool while the `Run` runs?

Reply:
Nvm, it seems that it will live till the end of Run. :D